xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/XrplClient.java

```diff
@@ -232,6 +239,80 @@ public FeeResult fee() throws JsonRpcClientErrorException {
     return jsonRpcClient.send(request, FeeResult.class);
   }
 
+  /**
+   * Get the ledger index of a tx result response. If not present, throw an exception.
+   *
+   * @return A string containing value of last validated ledger index.
+   */
+  private UnsignedInteger getMostRecentlyValidatedLedgerIndex() throws JsonRpcClientErrorException {
+    JsonRpcRequest request = JsonRpcRequest.builder()
+      .method(XrplMethods.LEDGER)
+      .addParams(LedgerRequestParams.builder().ledgerSpecifier(LedgerSpecifier.VALIDATED).build())
+      .build();
+    return jsonRpcClient.send(request, LedgerResult.class).ledgerIndexSafe().unsignedIntegerValue();
+  }
+
+  /**
+   * Check if the transaction is final on the ledger or not.
+   *
+   * @param transactionHash {@link Hash256} of the submitted transaction to check the status for.
+   * @param submittedOnLedgerIndex {@link LedgerIndex} on which the transaction with hash transactionHash
+   *        was submitted. This can be obtained from submit() response of the tx as validatedLedgerIndex.
+   * @return {@code true} if the {@link Transaction} is final/validated else {@code false}.
+   * @throws JsonRpcClientErrorException if {@code jsonRpcClient} throws an error.
+   */
+  public boolean isFinal(Hash256 transactionHash, LedgerIndex submittedOnLedgerIndex)
+    throws JsonRpcClientErrorException {
+    JsonRpcRequest request = JsonRpcRequest.builder()
+      .method(XrplMethods.TX)
+      .addParams(TransactionRequestParams.of(transactionHash))
+      .build();
+
+    TransactionResult response = jsonRpcClient.send(request, TransactionResult.class);
+    UnsignedInteger lastLedgerSequence = response.transaction().lastLedgerSequence()
+      .orElseThrow(() -> new IllegalStateException("Ledger result did not contain LastLedgerSequence"));
+
+    // condition 1: validated = true
+    if (response.validated()) {
+      String txResult = ((TransactionMetadata) response.metadata().get()).transactionResult();
+      if (txResult.equals("tesSUCCESS")) {
+        return true;
+      } else {
+        return false;
+      }
+    } else if (FluentCompareTo.is(getMostRecentlyValidatedLedgerIndex()).lessThan(lastLedgerSequence)) {
+      // condition 2: last ledger seq hasn't passed yet
+      // possible solution would be to check again after a while to get the updated validation status.
+      return false;
+    } else {
+      // condition 3: last ledger seq has passed.
+      Range<UnsignedLong> submittedToLast = Range.closed(UnsignedLong.valueOf(submittedOnLedgerIndex.toString()),
```

Comment:
Why not make this a `Range<UnsignedInteger>` so you don't have to do all this `UnsignedInteger` -> `UnsignedLong` conversion?

Reply:
Because, completeLedgerRanges is coming in as Range<UnsignedLong>. So, I have to convert either one of those.